src/server_manager/web_app/digitalocean_account.ts

```diff
@@ -121,7 +125,8 @@ export class DigitalOceanAccount implements digitalocean.Account {
 
   // Creates a DigitalOceanServer object and adds it to the in-memory server list.
   private createDigitalOceanServer(digitalOcean: DigitalOceanSession, dropletInfo: DropletInfo) {
-    const server = new DigitalOceanServer(digitalOcean, dropletInfo);
+    const server =
+        new DigitalOceanServer(`${this.id}:${dropletInfo.id}`, digitalOcean, dropletInfo);
```

Comment:
Maybe you can help me think through this a little bit.

This reminds me of a comment you mentioned a while back with account IDs. It got me wondering if they should exist in the model or the UI? And if they exist in the model – in what form?

I wonder if they should just be the IDs provided by the cloud providers (account ID - uuid for DO, server ID - droplet ID). If we do require them to be unique for the UI, we can create unique IDs in the UI models (ServerListEntry).

Or do we treat the account ID and server ID as product concepts and make them globally unique like you're doing?

Reply:
It all depends on how the ids are used.

If IDs are real-world ids, your model may need them. That's definitely a domain model concept. For example, a car rental may need your ID from the driver's license. You will want to expose that as a method. Though you may not actually want to identify a user by their driver's license, since that may actually change (expire, get a new one from another state). In that case you'd want a separate, system-generated ID for the customer.

In our case, we don't need to expose IDs anywhere. For the purpose of linking model entities, or linking model and the UI, we could have random ids that we generate at loading time, [like we do in the client](https://github.com/Jigsaw-Code/outline-client/blob/74dcb85b859a1871130d67292019b8cacdcffe43/src/www/app/outline_server.ts#L151). All we need is the id to be unique and stable during the app execution. That's not really a domain model concept, except that it represents the identity of the entity.

Actually, for the most part we can establish identity by leveraging object equality without even having IDs. I demonstrate that on https://github.com/Jigsaw-Code/outline-server/pull/858.

However, it turns out we need a stable id for the servers because we persist the last selected one. We can accomplish a stable server id without persistence by combining the account uuid and the droplet id, which is what I use in this PR. In theory I don't need the "do:" prefix, since DO says the account has a uuid, which is supposedly unique,  but the prefix is useful for debugging and clarity.

/cc @alalamav @JonathanDCohen who may be interested in the design discussion.

Reply:
Thanks for the explanation.

I went back and forth on this approach a few times. I think your solution of updating the Server#id to be unique is a good one.